### ISSUE TITLE
PETSc build script does not pass command-line arguments

### ISSUE THREAD

**Issue by joshuahansel** [idaholab/moose#12763]
## Rationale

Users should be able to pass command-line arguments to PETSc's configure script.

## Description

Unlike `update_and_rebuild_libmesh`, `update_and_rebuild_petsc` does not pass its command-line arguments to the corresponding submodule's `configure`.

## Impact

Greater functionality in script.

**Comment by joshuahansel** [comment]
@fdkong @lindsayad I think the solution is just to add `$*` add the end of the `./configure` line in `update_and_rebuild_petsc`, but I'd figure I'd try a PETSc submodule update as well. I ran into this because I was trying to build with PETSc master, and I got an error during the build (during `Compiling FBLASLAPACK`):
```
Cannot use SuperLU_DIST without enabling C++11, see --with-cxx-dialect=C++11
```
I then tried to pass `--with-cxx-dialect=C++11` to `update_and_rebuild_petsc`, and it didn't. After fixing `update_and_rebuild_petsc`, this works. Should we add this flag to the default list of flags passed to `configure`? 

I was going to go ahead and update the PETSc module with this change as well. It seems that the enum `KSP_DIVERGED_PCSETUP_FAILED` was removed; it suggested `KSP_DIVERGED_PC_FAILED` instead (would this be correct)? How would I guard this? The change seems to have occurred sometime within PETSc 3.10, so I would need know the subminor version at which the change occurred. Does PETSc have a summary of changes with each subminor version that I can view?
